libtsuba/include/katana/PreparedUpdate.h

```diff
@@ -0,0 +1,103 @@
+#ifndef KATANA_LIBTSUBA_KATANA_PREPAREDUPDATE_H_
+#define KATANA_LIBTSUBA_KATANA_PREPAREDUPDATE_H_
```

Comment:
This is actually only required because of a stale header in `StorageBackedArrowArray.h` it's not used anywhere.
That said; moving it seems fine since you're doing a lot of that anyway; can we put it in `libsupport` since it's not storage specific?